corehq/apps/ota/views.py

```diff
@@ -112,6 +112,12 @@ def app_aware_search(request, domain, app_id):
         case_search_criteria = CaseSearchCriteria(domain, case_type, criteria)
     except TooManyRelatedCasesError:
         return HttpResponse(_('Search has too many results. Please try a more specific search.'), status=400)
+    except CaseFilterError as e:
+        # This is an app building error, notify so we can track
+        notify_exception(request, str(e), details=dict(
+            exception_type=type(e),
+        ))
+        return HttpResponse(str(e), status=400)
```

Comment:
should this be 500 instead of 400? 400 would be for bad requests, but the request here is okay and HQ isn't able to generate the response.

Reply:
I don't feel that strongly about this, but I am mimicking the case type error a couple of lines up, which seems like the same kind of thing: the request has bad parameters that HQ doesn't know what to do with.

Reply:
Ya, that one seems to be okay with giving back 400, so that user should filter differently.  But if the app built is error-ring, we should not tell user to retry.

Also, confused why the app would error now? Isn't it already installed on the phone?